File: java_projects/glue_code/commons-validator/src/main/java/org/apache/commons/validator/routines/CurrencyValidator.java
package org.apache.commons.validator.routines;

import java.text.Format;
import org.apache.commons.validator.ContextInitializer;
import org.graalvm.polyglot.Value;

public class CurrencyValidator extends BigDecimalValidator {
  private static final char CURRENCY_SYMBOL = '\u00A4';
  private static final CurrencyValidator VALIDATOR = CurrencyValidator.CurrencyValidator1();
  private static final long serialVersionUID = -4201640771171486514L;
  private static Value clz =
      ContextInitializer.getPythonClass("/routines/CurrencyValidator.py", "CurrencyValidator");
  private Value obj;

  public CurrencyValidator(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  protected Object parse(String value, Format formatter) {
    //
    //
    // Object parsedValue = super.parse(value, formatter);
    // if (parsedValue != null || !(formatter instanceof DecimalFormat)) {
    // return parsedValue;
    // }
    //
    // DecimalFormat decimalFormat = (DecimalFormat) formatter;
    // String pattern = decimalFormat.toPattern();
    // if (pattern.indexOf(CURRENCY_SYMBOL) >= 0) {
    // StringBuilder buffer = new StringBuilder(pattern.length());
    // for (int i = 0; i < pattern.length(); i++) {
    // if (pattern.charAt(i) != CURRENCY_SYMBOL) {
    // buffer.append(pattern.charAt(i));
    // }
    // }
    // decimalFormat.applyPattern(buffer.toString());
    // parsedValue = super.parse(value, decimalFormat);
    // }
    // return parsedValue;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("parse", value, formatter).as(Object.class);
  }

  public static CurrencyValidator CurrencyValidator1() {
    //
    // return new CurrencyValidator(true, true);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("CurrencyValidator1").as(CurrencyValidator.class);
  }

  public CurrencyValidator(boolean strict, boolean allowFractions) {
    //
    // super(strict, CURRENCY_FORMAT, allowFractions);
    //

    this.obj = clz.invokeMember("__init__", strict, allowFractions);
  }

  public static BigDecimalValidator getInstance() {
    //
    // return VALIDATOR;
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("getInstance").as(BigDecimalValidator.class);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/commons-validator/src/main/org/apache/commons/validator/routines/CurrencyValidator.py
from __future__ import annotations
import io
import typing
from typing import *
import datetime
from src.main.org.apache.commons.validator.routines.AbstractNumberValidator import *
from src.main.org.apache.commons.validator.routines.BigDecimalValidator import *


class CurrencyValidator(BigDecimalValidator):

    __CURRENCY_SYMBOL: str = "\u00A4"

    __VALIDATOR: CurrencyValidator = CurrencyValidator.CurrencyValidator1()

    __serialVersionUID: int = -4201640771171486514

    def _parse(
        self, value: str, formatter: typing.Union[str, datetime.datetime]
    ) -> typing.Any:

        parsedValue = super()._parse(value, formatter)
        if parsedValue is not None or not isinstance(formatter, DecimalFormat):
            return parsedValue

        decimalFormat = formatter
        pattern = decimalFormat.toPattern()
        if pattern.find(self.__CURRENCY_SYMBOL) >= 0:
            buffer = []
            for i in range(len(pattern)):
                if pattern[i] != self.__CURRENCY_SYMBOL:
                    buffer.append(pattern[i])
            decimalFormat.applyPattern("".join(buffer))
            parsedValue = super()._parse(value, decimalFormat)
        return parsedValue

    @staticmethod
    def CurrencyValidator1() -> CurrencyValidator:

        return CurrencyValidator(True, True)

    def __init__(self, strict: bool, allowFractions: bool) -> None:

        super().__init__(strict, CURRENCY_FORMAT, allowFractions)

    @staticmethod
    def getInstance() -> BigDecimalValidator:

        if not hasattr(CurrencyValidator, "__VALIDATOR"):
            CurrencyValidator.__VALIDATOR = BigDecimalValidator()

        return CurrencyValidator.__VALIDATOR
